File: cvtheque-spring-boot/src/main/java/org/sid/cvthequespringboot/dtos/EvaluationDto.java
package org.sid.cvthequespringboot.dtos;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.sid.cvthequespringboot.entities.CriteriaEval;

import java.util.Date;
import java.util.List;

@Getter
@Setter
public class EvaluationDto {
    private Long id;

    private Date createdAt = new Date();

    private List<CriteriaEval> acceptCriteria;

    private List<CriteriaEval> rejectCriteria;

    private List<String> existWords;

    private List<String> noExistWords;

    private List<ScoringDto> scoring;
}

File: cvtheque-front-angular/src/app/Models/EvaluationDto.ts
//Evaluation
export interface EvaluationDto {
  id: number;
  createdAt: Date;
  acceptCriteria: CriteriaEval[];
  rejectCriteria: CriteriaEval[];
  existWords: string[];
  noExistWords: string[];
  scoring: ScoringDto[];
}
export interface CriteriaEval {
  id: number;
  createdAt: Date;
  status: Status;
  name: string;
  message: string;
}
export interface ScoringDto {
  profile: string;
  score: number;
  message: string;
  createdAt: Date;
}
export enum Status {
  ACCEPTED = 'ACCEPTED',
  REJECTED = 'REJECTED',
}
